npm/parseSelector.go

```diff
@@ -121,73 +121,170 @@ func sortSelector(selector *metav1.LabelSelector) {
 	selector.MatchExpressions = sortedReqs
 }
 
+// getSetNameForMultiValueSelector takes in label with multiple values without operator
+// and returns a new 2nd level ipset name
+func getSetNameForMultiValueSelector(key string, vals []string) string {
+	rtStr := key
+	for _, val := range vals {
+		rtStr = util.GetIpSetFromLabelKV(rtStr, val)
+	}
+	return rtStr
+}
+
 // HashSelector returns the hash value of the selector.
 func HashSelector(selector *metav1.LabelSelector) string {
 	sortSelector(selector)
 	return util.Hash(fmt.Sprintf("%v", selector))
 }
 
-// parseSelector takes a LabelSelector and returns a slice of processed labels, keys and values.
-func parseSelector(selector *metav1.LabelSelector) ([]string, []string, []string) {
+// flattenNameSpaceSelector will help flatten multiple NameSpace selector match Expressions values
+// into multiple label selectors helping with the OR condition.
+func FlattenNameSpaceSelector(nsSelector *metav1.LabelSelector) []metav1.LabelSelector {
+	//egress section
+	// for _, egressRule := range nsSelector {
+	// 	log.Logf("%+v", egressRule)
+	// }
+	if nsSelector == nil {
+		return []metav1.LabelSelector{*nsSelector}
```

Comment:
Not sure why it includes *nsSelector. Any reasons? Will be the same to just return nil or empty slices.

Reply:
The way we designed, FlattenNSSelector will return a list of selectors, so it can be looped over, thats why for an nill or empty selector, i am just returning the original selector as a list.

Reply:
Is "[]metav1.LabelSelector{*nsSelector} =>  []metav1.LabelSelector{}" different if nsSelector is nil?

Reply:
Luckily, no one from translate.go calls this FlattenNameSpaceSelector function with **nil nsSelector** value in current NPM. 
However, we are not sure who calls this function in the future. So, it is good to check nil value at the first place.

However, in that case, it should be below code if "nsSelector" is nil. 
Otherwise, it will be panic. We cannot dereference nil value.
```go
return []metav1.LabelSelector{}
```

Reply:
I think this is a good lesson. We should add various and edge cases in UTs. 
Suggest adding UT for nil test for this function.